newsroom/auth/views.py

```diff
@@ -87,3 +116,61 @@ def _modify_user_data(new_user):
     new_user.pop('company', None)
     new_user.pop('occupation', None)
     new_user.pop('company_size', None)
+    new_user.pop('csrf_token', None)
+
+
+def _add_token_data(user):
+    user['token'] = get_random_string()
+    user['token_expiry_date'] = utcnow() + timedelta(days=app.config['VALIDATE_ACCOUNT_TOKEN_TIME_TO_LIVE'])
+
+
+@blueprint.route('/validate')
+def validate_account():
+    token = flask.request.args['token']
+    if token:
```

Comment:
maybe this route should be `/validate/<token>` so there is no need for this `if token:`